File: api/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/api/utils/featuretoggling/FeatureToggle.java
package no.nav.sbl.dialogarena.modiabrukerdialog.api.utils.featuretoggling;

import static java.lang.Boolean.valueOf;
import static java.lang.System.getProperty;
import static java.lang.System.setProperty;

public class FeatureToggle {

    public static boolean visFeature(Feature feature) {
        return valueOf(System.getProperty(feature.propertyKey, String.valueOf(feature.defaultValue)));
    }

    public static void toggleFeature(Feature feature) {
        setProperty(feature.propertyKey, "true");
    }

    public static void disableFeature(Feature feature) {
        setProperty(feature.propertyKey, "false");
    }

}

File: api/src/main/java/no/nav/sbl/dialogarena/modiabrukerdialog/api/utils/featuretoggling/FeatureToggle.kt
package no.nav.sbl.dialogarena.modiabrukerdialog.api.utils.featuretoggling

import java.lang.System.getProperty
import java.lang.System.setProperty

fun visFeature(feature: Feature) = getProperty(feature.propertyKey, feature.defaultValue.toString())!!.toBoolean()

fun enableFeature(feature: Feature) { setProperty(feature.propertyKey, "true") }

fun disableFeature(feature: Feature) { setProperty(feature.propertyKey, "false") }
